lacci/lib/shoes/widgets/oval.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Shoes
+  class InvalidAttributeValueError < StandardError; end
+
+  class Oval < Shoes::Widget
+    include Shoes::Log
+    display_properties :left, :top, :width, :height, :center, :fill, :stroke, :strokewidth
+
+    def initialize(left = nil, top = nil, width = nil, height = nil, **options)
+      @left = convert_to_integer(left || options.fetch(:left), "left")
+      @top = convert_to_integer(top || options.fetch(:top), "top")
+      @width = convert_to_float(width || options.fetch(:radius) * 2, "width")
+      @height = if height.nil?
+        @width
+      else
+        convert_to_float(height, "height")
+      end
+      @center = options.fetch(:center, false)
+      @fill = options.fetch(:fill, nil)
+      @stroke = options.fetch(:stroke, nil)
+      @stroke_width = options.fetch(:strokewidth, 2)
+
+      super()
```

Comment:
If you call "super" rather than "super" you should be able to skip some of those assignments for display properties by using real keywords. If there's a fill: keyword and you pass it to the Shoes::Widget constructor, it will assume its value becomes the value of the display property called fill and the instance var fill.

Reply:
Although as I write this, it might make more sense to do cascading fallbacks there -- fill if specified, falling back to the display context. So maybe we don't have many properties that can just be set by the default constructor.